### PR TITLE
update reify and add cache to .gitignore, closes #63

### LINKED ISSUE #63: Error trying to execute `npm run repl`
Hello everyone,

I'm trying to use the boilerplate, but I got the error below:

```shell
o0x41e@lastzion ~/Development/js/graphql-dataloader-boilerplate (master)                                              
 ~>npm run repl                                                                                                       

> graphql-dataloader-boilerplate@0.0.1 repl /home/o0x41e/Development/js/graphql-dataloader-boilerplate                
> nodemon --config ./repl/nodemon.json ./repl.js --exec babel-node                                                    

[nodemon] 1.12.1             
[nodemon] watching: *.*      
[nodemon] starting `babel-node ./repl.js`                  
/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node/caching-compiler.js:21             
    throw e;                 
    ^                        

SyntaxError: Unexpected token (7:35) while processing file: /home/o0x41e/Development/js/graphql-dataloader-boilerplate/src/auth.js
    at Parser.pp$4.raise (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:2549:13)
    at Parser.pp.unexpected (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:635:8)
    at Parser.pp.expect (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:629:26)
    at Parser.pp$2.parseBindingList (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:1591:17)
    at Parser.pp$1.parseFunctionParams (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:1177:22)
    at Parser.pp$1.parseFunction (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:1164:8)
    at Parser.pp$1.parseFunctionStatement (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:883:15)
    at Parser.pp$1.parseStatement (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:787:19)
    at parseExportNamedDeclaration (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/lib/parsers/acorn-extensions/export.js:115:29)
    at Parser.parseExport (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/lib/parsers/acorn-extensions/export.js:31:12)
    at Parser.pp$1.parseStatement (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:777:69)
    at Parser.pp$1.parseTopLevel (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:684:23)
    at Parser.parse (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node_modules/acorn/dist/acorn.js:541:15)
    at parse (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/lib/parsers/acorn.js:17:17)
    at exports.compile (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/lib/compiler.js:31:15)
    at compileAndWrite (/home/o0x41e/Development/js/graphql-dataloader-boilerplate/node_modules/reify/node/caching-compiler.js:31:18)
[nodemon] app crashed - waiting for file changes before starting...  
```

I already tried to update the babel using `npm install babel`:
```
o0x41e@lastzion ~/Development/js/graphql-dataloader-boilerplate (master) 
 ~>npm run repl

> graphql-dataloader-boilerplate@0.0.1 repl /home/o0x41e/Development/js/graphql-dataloader-boilerplate
> nodemon --config ./repl/nodemon.json ./repl.js --exec babel-node

[nodemon] 1.12.1
[nodemon] watching: *.*
[nodemon] starting `babel-node ./repl.js`
You have mistakenly installed the `babel` package, which is a no-op in Babel 6.
Babel's CLI commands have been moved from the `babel` package to the `babel-cli` package.

    npm uninstall babel
    npm install --save-dev babel-cli

See http://babeljs.io/docs/usage/cli/ for setup instructions.
[nodemon] app crashed - waiting for file changes before starting...
```

I did follow the recommended steps and the initial error come back. Any help would be great (maybe some bug in the repl ?)

Some useful data:

```
# I'm using Linux (Slackware64 current)
o0x41e@lastzion ~/Development/js/graphql-dataloader-boilerplate (master) 
 ~>npm version
{ 'graphql-dataloader-boilerplate': '0.0.1',
  npm: '3.10.10',
  ares: '1.10.1-DEV',
  http_parser: '2.7.0',
  icu: '58.2',
  modules: '48',
  node: '6.11.2',
  openssl: '1.0.2l',
  uv: '1.11.0',
  v8: '5.1.281.103',
  zlib: '1.2.11' }
o0x41e@lastzion ~/Development/js/graphql-dataloader-boilerplate (master) 
 ~>npm --version
3.10.10
```

Thank you.

### PR DESCRIPTION
**Pull request by ronal2do**

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import jwt from 'jsonwebtoken';
 import { User } from './model';
 import { jwtSecret } from './config';
 
-export async function getUser(token: string) {
```
Comment (jgcmarins) [inline src/auth.js]: can we keep type?

File: src/auth.js [hunk 2]
```diff
@@ -20,10 +20,6 @@ export async function getUser(token: string) {
   }
 }
 
-type UserType = {
-  _id: string,
-}
-
-export function generateToken(user: UserType) {
```
Comment (jgcmarins) [inline src/auth.js]: can we also keep types here?

File: src/config.js [hunk 3]
```diff
@@ -9,24 +9,24 @@ dotenvSafe.load({
   sample: root('.env.example'),
 });
 
-const ENV = ((process.env: any): {
```
Comment (jgcmarins) [inline src/config.js]: can we keep this const?

Comment (ronal2do) [reply src/config.js]: Babel transform is not working with flow types.

And I need help for to verify this problem

plugins: [['transform-flow-strip-types']],

Comment (jgcmarins) [reply src/config.js]: try this version: `"babel-plugin-transform-flow-strip-types": "^6.22.0",` instead

File: .babelrc [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-    "stage-0",
     ["env", {
       "targets": {
-        "node": "current"
+        "node": "current",
+        "browsers": ["last 2 versions", "safari >= 7"]
       }
-    }]
+    }],
+    "stage-0"
```
Comment (jgcmarins) [inline .babelrc]: I guess `.babelrc` can be something like this:
```
{
  "presets": [
    "flow",
    ["env", {
      "targets": {
        "node": "current"
      }
    }]
  ],
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    "transform-export-extensions",
    "transform-async-to-generator"
  ]
}
```

and then you can add the following babel plugins:
```
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-class-properties": "^6.23.0",
    "babel-plugin-transform-export-extensions": "^6.22.0",
    "babel-plugin-transform-flow-strip-types": "^6.22.0",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-preset-env": "^1.6.1",
```

File: src/config.js (hunk 3 above)
Comment (sibelius) [reply src/config.js]: Use flow preset

**Comment by ronal2do** [comment]
the esm solved all problems

**Pull request merged**